### ISSUE TITLE
should emit `invalid-metaclass` for callable with wrong signature

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3733]
### Summary

https://docs.astral.sh/ty/reference/rules/#invalid-metaclass

```py
def f(): ...

class B(metaclass=f): ... # no error
```
[Playground](https://play.ty.dev/d767e9dc-70ef-4700-8570-e2aebeca10d7)

### Version

267af002f

**Comment by carljm** [comment]
Yeah it looks like this validation was marked as a TODO when the rule landed, a long time ago, and we just never got back to add it. We should just add it. I think https://github.com/astral-sh/ruff/pull/24550 might do this? Not sure if there are blockers to moving that forward, cc @AlexWaygood .

**Comment by AlexWaygood** [comment]
> Not sure if there are blockers to moving that forward, cc [@AlexWaygood](https://github.com/AlexWaygood) .

just prioritisation
